### DISCUSSION TITLE
how do i fix an entity randomly spinning around whenever its walking?

### DISCUSSION THREAD

**Discussion by ANightDazingZoroark** [FabricMC/fabric-api#2439] [Mod Dev Support]
basically what it says on the title. figured i'd post my code here for reference: https://github.com/ANightDazingZoroark/Prehistoric-Rift-Mod/blob/remake/src/main/java/net/anightdazingzoroark/rift/entities/creatures/TyrannosaurusEntity.java

also, video showing off the problem https://youtu.be/D6Q8ur10YIE

**Comment by warjort** [comment]
If somebody does know, can they also tell Mojang. :-)
https://bugs.mojang.com/browse/MC-150293

**Reply by ANightDazingZoroark** [reply]
its not climbing scaffolding, its just walking around a walkable surface. also, the spinning around part is only for a quick second

**Comment by ANightDazingZoroark** [comment]
Heres a video showing it off just in case https://youtu.be/D6Q8ur10YIE

**Comment by Sn0wix** [comment]
I have the same problem :D

**Comment by FilipeGamer12** [comment]
do you found the fix?

**Comment by Sn0wix** [comment]
Yes, and it's pretty simple. Mojang's path finding for some reason doesn't work with bigger hitboxes. Mowzie's mobs developers made their own path finding system and I ported it to fabric. You can find it here: 
https://github.com/Sn0wix/The-World-of-Dragons-Mod/tree/master/src/main/java/net/sn0wix_/worldofdragonsmod/common/entity/ai. It's an older version of minecraft but you should be able to update it.

And you also have to replace the old navigation:
```
@Override
    protected EntityNavigation createNavigation(World world) {
        return new MMPathNavigateGround(this, world);
    }
```
